sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -343,12 +343,37 @@ export class EventHubSender extends LinkEntity {
   }
   /**
    * Returns maximum message size on the AMQP sender link.
-   * @ignore
+   * @param abortSignal An implementation of the `AbortSignalLike` interface to signal the request to cancel the operation.
+   * For example, use the &commat;azure/abort-controller to create an `AbortSignal`.
    * @returns Promise<number>
+   * @throws {AbortError} Thrown if the operation is cancelled via the abortSignal.
    */
-  async getMaxMessageSize(): Promise<number> {
-    try {
-      if (!this.isOpen()) {
+  async getMaxMessageSize(abortSignal?: AbortSignalLike): Promise<number> {
+    if (this.isOpen()) {
+      return this._sender!.maxMessageSize;
+    }
+    return new Promise<number>(async (resolve, reject) => {
+      const rejectOnAbort = () => {
+        const desc: string = `[${this._context.connectionId}] The create batch operation has been cancelled by the user.`;
+        log.error(desc);
+        const error = new AbortError(`The create batch operation has been cancelled by the user.`);
+        reject(error);
+      };
+
+      const onAbort = () => {
+        abortSignal!.removeEventListener("abort", onAbort);
```

Comment:
`abortSignal` is optional, it could be undefined. Why are we assuming that it wont be?

Reply:
Updated:
```javascript
 if (abortSignal) {
     abortSignal.removeEventListener("abort", onAbort);
 }
```